Moved to absolute path for playbook_path... and playbook.basedir

dingus9 opened the pull request:
The goal is to mirror the absolute path like what is in {{inventory_dir}} with {{playbook_dir}} instead of the relative path usually ".". I just wrapped the two primary places based_dir is used in os.path.realpath().

I tested the code with a few ansible runs however I didn't see any unit tests for playbooks specifically... If you point me to the test sweet I should run for this, I'll run it and update this PR.

Thanks

dingus9 committed:
Moved to absolute path for playbook_path... and playbook.basedir

mpdehaan commented:
I'm ok with calling the realpath stuff here but I'm very unclear about the "or" additions.

Can you explain the intent of these in greater detail?

dingus9 commented:
I added the or in play.py but the or was already in the **init**.py. Actually looking at it I'm thinking it should just ignore the basedir parameter and trust what is provided by the playbook object. I changed it to just accepting the internal property playbook.basedir since that should always be the basedirectory of the playbook.

How does that look

mpdehaan commented:
# Revision Request

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We'd like to see a few things tweaked if you don't mind.  If you can help resolve these items, we'd greatly appreciate it:
- this code change currently breaks "make tests", if you can update the code such that it passes, or fix the tests we would appreciate it

Just as a quick reminder of things, this is a really busy project.  We have over 800 contributors and to manage the queue effectively
we assign things a priority between P1 (highest) and P5.  We'd like to thank you very much for your time!
We'll work things in priority order, so just wanted you to be aware of the queue and know we haven't forgot about you!

We will definitely see your comments on this issue when reading this ticket, but may not be able to reply promptly.  You may also wish to join one of our two mailing lists
which is very active:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Thank you once again for this and your interest in Ansible!

mpdehaan commented:
Closing this due to lack of activity int he above, please feel free to resubmit.  Thanks!
